ayon_server/events/enroll.py

```diff
@@ -133,24 +134,33 @@ async def enroll_job(
 
         # Target event does not exist yet. Create a new one
         new_hash = hash_data((target_topic, row["source_id"]))
-        new_id = await EventStream.dispatch(
-            target_topic,
-            sender=sender,
-            hash=new_hash,
-            depends_on=row["source_id"],
-            user=user_name,
-            description=description,
-            finished=False,
-        )
+        try:
+            new_id = await EventStream.dispatch(
+                target_topic,
+                sender=sender,
+                hash=new_hash,
+                depends_on=row["source_id"],
+                user=user_name,
+                description=description,
+                finished=False,
+            )
+
+        except ConstraintViolationException:
+            # for some reason, the event already exists
+            # most likely because another worker took it
+
```

Comment:
You could just set `new_id = None` under exception handling, and keep resto of the code as was.